modules/common/serdes/examples/schemaless_example.cpp

```diff
@@ -0,0 +1,136 @@
+//=================================================================================================
+// Copyright (C) 2024 GRAPE Contributors
+//=================================================================================================
+
+#include <cstdlib>
+#include <fstream>
+#include <optional>
+#include <print>
+#include <vector>
+
+#include <flatbuffers/flexbuffers.h>
+
+#include "grape/serdes/serdes.h"
+
+//-------------------------------------------------------------------------------------------------
+struct State {
+  std::string name;
+  double timestamp{};
+  std::array<double, 3> position{};
+};
+
+namespace {
+//-------------------------------------------------------------------------------------------------
+auto serialize(const State& state) -> std::vector<std::uint8_t> {
+  flexbuffers::Builder fbb;
+  fbb.Map([&]() {
+    fbb.String("name", state.name);
+    fbb.Double("timestamp", state.timestamp);
+    fbb.Vector("position", [&]() {
+      for (const auto& p : state.position) {
+        fbb.Double(p);
+      }
+    });
+  });
+  fbb.Finish();
+
+  return fbb.GetBuffer();
```

Comment:
Is this a new vector being moved out? Check memory safety, memory allocation, memory copies this results in